modules/auxiliary/scanner/http/softing_sis_login.rb

```diff
@@ -0,0 +1,146 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'metasploit/framework/login_scanner/softing_sis'
+require 'metasploit/framework/credential_collection'
+
+class MetasploitModule < Msf::Auxiliary
+
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Auxiliary::AuthBrute
+  include Msf::Auxiliary::Report
+  include Msf::Auxiliary::Scanner
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Softing Secure Integration Server Login Utility',
+        'Description' => %q{
+          This module will attempt to authenticate to a Softing Secure Integration Server.
+        },
+        'Author' => [ 'Imran E. Dawoodjee <imrandawoodjee.infosec[at]gmail.com>' ],
+        'License' => MSF_LICENSE,
+        'Notes' => {
+          'Stability' => [],
+          'Reliability' => [],
+          'SideEffects' => []
+        },
+        'DefaultOptions' => {
+          'RPORT' => 8099,
+          'SSL' => false,
+          'SSLVersion' => 'TLS1'
+        }
+      )
+    )
+
+    deregister_options('PASSWORD_SPRAY')
+
+    # credentials are "admin:admin" by default
+    register_options(
+      [
+        OptString.new('USERNAME', [false, 'The username to specify for authentication. Not required if SIGNATURE is set.', 'admin']),
+        OptString.new('PASSWORD', [false, 'The password to specify for authentication. Not required if SIGNATURE is set.', 'admin'])
+      ]
+    )
+  end
+
+  def scanner(ip)
+    @scanner ||= lambda {
+      cred_collection = Metasploit::Framework::CredentialCollection.new(
+        blank_passwords: datastore['BLANK_PASSWORDS'],
+        pass_file: datastore['PASS_FILE'],
+        password: datastore['PASSWORD'],
+        user_file: datastore['USER_FILE'],
+        userpass_file: datastore['USERPASS_FILE'],
+        username: datastore['USERNAME'],
+        user_as_pass: datastore['USER_AS_PASS']
+      )
+
+      return Metasploit::Framework::LoginScanner::SoftingSIS.new(
+        configure_http_login_scanner(
+          host: ip,
+          port: datastore['RPORT'],
+          cred_details: cred_collection,
+          stop_on_success: datastore['STOP_ON_SUCCESS'],
+          bruteforce_speed: datastore['BRUTEFORCE_SPEED'],
+          connection_timeout: 5
+        )
+      )
+    }.call
+  end
+
+  def report_good_cred(ip, port, result)
```

Comment:
if one of the suggestions below is used then this can become:
```suggestion
  def report_good_cred(result)
```

Reply:
Also fixed in 2f08cf6c46407d6a70f9c194679490f3ec7ea951.